packages/wpcom-checkout/src/introductory-offer.ts

```diff
@@ -106,3 +106,25 @@ export function getIntroductoryOfferIntervalDisplay(
 	}
 	return text;
 }
+
+export function getItemIntroductoryOfferDisplay( product: ResponseCartProduct ) {
+	if ( product.introductory_offer_terms?.reason ) {
+		const text = translate( 'Order not eligible for introductory discount' );
```

Comment:
This is _probably_ fine, but I do remember that there's some downside with using `translate()` directly rather than passing it through `useTranslate()`/`localize()`... I think it might cause situations where the translation does not get the correct locale (otherwise we could just import `translate` everywhere instead of needing those constructs). How much do you need to not pass `translate` into this function?

Reply:
For the record I think that this might be the advantage of the dynamic translate method over the static one:

> the resulting component is also reactive, i.e., it gets rerendered when the `i18n` locale changes and the state emitter emits a `change` event.

I don't know how often that happens, but I just wanted to note it since importing `translate` directly here is a considerable change in behavior that might be outside the scope of what you're working on.

Reply:
I understand your assessment, and especially in the context of the checkout, you might be right that it's better to err on the side of caution. 

It's not necessary to import `translate` instead of passing it as an argument, but it seemed more convenient. This approach is also already being used in `packages/wpcom-checkout/src/checkout-labels.ts` (a neighbouring file).

After doing some digging, I realise that the "unconnected" `translate` might be problematic when loading translations (i.e. it won't render the new translation because it can't update the React state). Not sure how often that becomes an issue, but let's stick with the approach we have! Thanks for being diligent, @sirbrillig 😊